druid/src/widget/parse.rs

```diff
@@ -38,7 +43,9 @@ impl<T> Parse<T> {
 impl<T: FromStr + Display + Data, W: Widget<String>> Widget<Option<T>> for Parse<W> {
     fn event(&mut self, ctx: &mut EventCtx, event: &Event, data: &mut Option<T>, env: &Env) {
         self.widget.event(ctx, event, &mut self.state, env);
-        *data = self.state.parse().ok();
+        if self.state != self.prev_state {
```

Comment:
Would it make more sense to have `self.prev_state = self.state;` just before passing state to the inner widget here, and then scrapping lines 68-70? This would still ensure that the widget is updated properly with `prev_state` and `state`, while not setting the state via `data.to_string()`

Reply:
Thanks for checking this. There are some frustrating limitations to using `parse()` for this, instead of some dedicated validation/transformation mechanism, but I hope I have the main cases covered now!